daphne/access.py

```diff
@@ -8,7 +10,18 @@ class AccessLogGenerator:
     """
 
     def __init__(self, stream):
-        self.stream = stream
+        if stream:
+            logger.propagate = False
+            handler = logging.StreamHandler(stream)
+            formatter = logging.Formatter(
+                '%(host)s %(ident)s %(user)s [%(asctime)s] "%{request} %(message)s" '
+                "%(status)s %(length)s"
+                "%d/%b/%Y:%H:%M:%S"
+            )
+            handler.setFormatter(fmt=formatter)
+            logger.addHandler(handler)
+        else:
+            logger.addHandler(logging.NullHandler)
```

Comment:
I'm not super familiar with how the `logging` module works, but why are these lines needed? What happens if they are removed?

Reply:
My understanding is that the access log would be written to the default handler (likely console).  If I am understanding the cli correctly, the access log is not written unless explicitly enabled,

Reply:
Shouldn't we just set `propagate = False` unconditionally then? (I totally might be misunderstanding how this all works, please bear with me)

Reply:
I'll need to dig in after I recreate a testing environment.  Looks like some other tweaks may be needed